components/ModEditHealthcareProfessionalSection.vue

```diff
@@ -0,0 +1,279 @@
+                name="Locales"
+                multiple
+                class="mb-5 px-3 py-3.5 w-96 h-32 bg-secondary-bg rounded-lg border border-primary-text-muted
+                            text-primary-text text-sm font-normal font-sans placeholder-primary-text-muted"
+            >
+                <option
+                    v-for="(locale, index) in localeOptions"
+                    :key="`${locale}-${index}`"
+                    :value="locale"
+                >
+                    {{ locale }}
+                </option>
+            </select>
+        </div>
+    </div>
+</template>
+
+<script lang="ts" setup>
+import { onMounted } from 'vue'
+import { ModerationScreen, useModerationScreenStore } from '~/stores/moderationScreenStore'
+import { useHealthcareProfessionalsStore } from '~/stores/healthcareProfessionalsStore'
+import { Locale, type LocalizedNameInput, Insurance, Degree, Specialty } from '~/typedefs/gqlTypes'
+import { multiSelectWithoutKeyboard } from '~/utils/multiSelectWithoutKeyboard'
+import SVGTrashCan from '~/assets/icons/trash-can.svg'
+import SVGProfileIcon from '~/assets/icons/profile-icon.svg'
+
+const moderationScreenStore = useModerationScreenStore()
+const healthcareProfessionalsStore = useHealthcareProfessionalsStore()
+// less clear name necessary because of filter rules on line length (max = 130)
+const professionalNameArray = healthcareProfessionalsStore.healthcareProfessionalSectionFields.healthcareProfessionalNameArray
+const sectionFields = healthcareProfessionalsStore.healthcareProfessionalSectionFields
+
+// Creating the necessary parts for the multi-select
+const insuranceOptions = Object.values(Insurance) as Insurance[]
+const extractInsuranceOptions = (option: HTMLOptionElement): Insurance => option.value as Insurance
+const degreeOptions = Object.values(Degree) as Degree[]
+const extractDegreeOptions = (option: HTMLOptionElement): Degree => option.value as Degree
+const specialtyOptions = Object.values(Specialty) as Specialty[]
+const extractSpecialtyOptions = (option: HTMLOptionElement): Specialty => option.value as Specialty
+const localeOptions = Object.values(Locale) as Locale[]
+const extractLocaleOptions = (option: HTMLOptionElement): Locale => option.value as Locale
+
+const handleLocalizedName = () => {
+    const localizedNameToAdd: LocalizedNameInput = {
+        firstName: sectionFields.localizedFirstName.value,
+        lastName: sectionFields.localizedLastName.value,
+        locale: sectionFields.nameLocale.value || Locale.EnUs,
+        middleName: sectionFields.localizedMiddleName.value
+    }
+
+    if (localizedNameToAdd.firstName && localizedNameToAdd.lastName) {
+        professionalNameArray.value.push(localizedNameToAdd)
+        sectionFields.localizedFirstName.value = ''
+        sectionFields.localizedLastName.value = ''
+        sectionFields.localizedMiddleName.value = ''
+        sectionFields.nameLocale.value = Locale.EnUs
+    }
+}
```

Comment:
What do we want to happen here if the `localizedNameToAdd` is missing these two values? Currently, it looks like the request will fail quietly. We may want to consider showing a validation error.

Reply:
@theyokohamalife do we have an example of how we usually implement this? Is it just a simple case of importing a validation check from our utils file and directly adding it to the relevant ModInputFields?